opendevin/core/config.py

```diff
@@ -348,22 +348,9 @@ def set_attr_from_env(sub_config: Any, prefix=''):
                 nested_sub_config = getattr(sub_config, field_name)
 
                 # the agent field: the env var for agent.name is just 'AGENT'
-                if field_name == 'agent' and env_var_name in env_or_toml_dict:
+                if field_name == 'agent' and 'AGENT' in env_or_toml_dict:
```

Comment:
reason for this revert? and the removal of exec type?

Reply:
1) It's okay both ways, this isn't much of a change either way. I just find stating it explicitly is a little easier to read, as in, no need to look around to see to what exactly do these lines apply.

This case is a special case, for when we had AGENT env var. It needs to be hardcoded no matter how we write it, just one way is slightly hiding it and the other is showing it plainly.

2) Exec box was removed by another PR today. Check the codebase, it went poof 😅

Reply:
but in the next line `env_or_toml_dict[env_var_name]` is used. so inconsistent.

Reply:
We can make it better at the next pass, when we'll have more vars without any prefix.